src/request_items.rs

```diff
@@ -31,7 +39,7 @@ impl RequestItems {
         count
     }
 
-    pub fn headers(&self) -> (HeaderMap<HeaderValue>, Vec<HeaderName>) {
+    pub fn headers(&self, session: &Option<Session>) -> (HeaderMap<HeaderValue>, Vec<HeaderName>) {
```

Comment:
Any particular reason to use `&Option<Session>` here instead of `Option<&Session>`?

Reply:
No particular reason, I'm afraid I just wanted to use a ref on the function argument, the built-in mechanism to do this was lost on me. This is really helpful (as every comment on this PR), thanks.